### PR TITLE
Implement Applicant Table

### PR DESCRIPTION
**Pull request by eevanwong**
### Implementation description
- This PR contains the hard coded student table for the admin page. Where we will view all of the applicants that applied during  recruitment season.
- It takes in an array of Students/Applicants which are displayed on each row with their name, email, resume link, and details link.
image

### Steps to test
1. Go to /admin
2. Login
3. View table (I left some mock data for easier testing)

### What should reviewers focus on?
- Had a bit of a difficult time styling table rows and hyperlinked stuff, there are global styles that are affecting my ability to change certain styles like the <a> hyperlinks.

### Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: components/common/Table.tsx
Comment (lambo-liu) [inline components/common/Table.tsx:1]: Let's rename this file and component to `ApplicationsTable` and move it to /components/admin/

File: components/common/Table.tsx
Comment (lambo-liu) [inline components/common/Table.tsx:3]: nit: Use type instead of interface here

File: components/common/Table.tsx
Comment (lambo-liu) [inline components/common/Table.tsx:39]: className="text-blue-100"

File: components/common/Table.tsx
Comment (lambo-liu) [inline components/common/Table.tsx:44]: Can just make this "Resume" so it doesn't take up two lines

File: components/common/Table.tsx
Comment (lambo-liu) [inline components/common/Table.tsx:48]: className="text-blue-100"

File: components/common/Table.tsx
Comment (lambo-liu) [inline components/common/Table.tsx:32]: Remove `border-indigo-500`?

File: components/common/Table.tsx
Comment (lambo-liu) [inline components/common/Table.tsx:18]: We can remove this div/containment as this component will be used in another container

File: components/common/Table.tsx
Comment (lambo-liu) [inline components/common/Table.tsx:22]: nit: Use `border-charcoal-400`

File: components/common/Table.tsx
Comment (lambo-liu) [reply components/common/Table.tsx:32]: Probably use `border-charcoal-400` instead

**Review by lambo-liu — CHANGES_REQUESTED** [review]
Just some styling changes, awesome stuff!

File: pages/admin/index.tsx
Comment (albertlai431) [inline pages/admin/index.tsx:12]: Looks like the table is at the very top after Abu's changes got merged, could we move it to after the `Export CSV` button?

File: components/admin/ApplicationsTable.tsx
Comment (albertlai431) [inline components/admin/ApplicationsTable.tsx:13]: nit but should be `students` cuz camel case

File: components/admin/ApplicationsTable.tsx
Comment (albertlai431) [inline components/admin/ApplicationsTable.tsx:4]: we're using uuid's that are strings so would be better to make this type `string`

File: components/admin/ApplicationsTable.tsx
Comment (albertlai431) [inline components/admin/ApplicationsTable.tsx:48]: we can just do `https://uwblueprint.org/admin/student-details/${app.id}` since that's how we have links implemented - in that case we can also just remove the `detailsLink` property from the `Student` type

**Review by albertlai431 — CHANGES_REQUESTED** [review]
Looking good, just a couple of small nits!

File: components/admin/ApplicationsTable.tsx
Comment (eevanwong) [reply components/admin/ApplicationsTable.tsx:4]: Ok, what will the uuid be used for?

File: pages/admin/index.tsx
Comment (albertlai431) [inline pages/admin/index.tsx:34]: It's currently rendering like: 
image

`ApplicationTable` should be after the first `</div>` tag

**Review by albertlai431 — APPROVED** [review]
Great work, just one last thing! Make sure to get approval from @lambo-liu as well

File: pages/admin/index.tsx
Comment (lambo-liu) [reply pages/admin/index.tsx:34]: Please fix

Comment (eevanwong) [reply pages/admin/index.tsx]: image

Changed location of component.

Comment (eevanwong) [reply pages/admin/index.tsx]: @lambo-liu

**Review by lambo-liu — APPROVED** [review]

**Pull request merged**
